src/Modal/index.jsx

```diff
@@ -139,21 +139,17 @@ class Modal extends React.Component {
   }
 
   renderButtons() {
-    return this.props.buttons.map((button, i) => {
-      let buttonElement = button;
-      let buttonProps = button.props;
-
-      if (button.type !== Button) {
-        buttonProps = button;
-      }
-
-      buttonElement = (<Button
-        {...buttonProps}
-        key={i}
-        onKeyDown={this.handleKeyDown}
-      />);
-
-      return buttonElement;
+    return this.props.buttons.map((button) => {
+      // button is either a Button component that we want clone or a set of props
+      const buttonProps = button.type === Button ? button.props : button;
```

Comment:
Why clone at all? I feel like this adds the kind of react hierarchy-indirection that makes me nervous after ADR-2 in this repo https://github.com/edx/paragon/blob/master/docs/decisions/0002-react-i18n.rst

Reply:
It seems that this was made to handle these two different ways of providing buttons. I agree we shouldn't do what we are here, but for the sake of backward compatibility I didn't want to change anything. This was more just cleaning up some code that I found hard to read.

Reply:
My jaw would drop if any downstream users of this component relied on the behavior of us not using the same buttons they pass in, but prop-clones of the same.

But I can understand the desire to push off this concern; should I make a note of it to drop in the backlog once we have one set up?

Reply:
Yes please do.

Reply:
This ended up being mooted in https://github.com/edx/paragon/pull/435 by the elimination of these buttons from the modal component's prop interface. Marking as resolved.